diff --git a/dist/xmldoc.js b/dist/xmldoc.js
index ca7b6a9755399cd328d20b9a0b2c9b43adb82051..36982c9654783fa37845fb3d922a6d53858b14e2 100644
--- a/dist/xmldoc.js
+++ b/dist/xmldoc.js
@@ -108,6 +108,15 @@ class XmlElement {
      * @param parser Optional SAX parser instance with position information
      */
     constructor(tag, parser) {
+        // If you didn't hand us a parser (common case) see if we can grab one
+        // from the current execution stack.
+        if (!parser && delegates.length) {
+          var delegate = delegates[delegates.length - 1];
+
+          if (delegate.parser) {
+            parser = delegate.parser;
+          }
+        }
         this.type = "element";
         this.name = tag.name;
         this.attr = tag.attributes;
